internal/services/containers/container_registry_cache_rule_resource_test.go

```diff
@@ -0,0 +1,167 @@
+package containers_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ContainerRegistryCacheRuleResource struct{}
+
+func TestAccContainerRegistryCacheRuleName_validation(t *testing.T) {
```

Comment:
This `validate` function is already tested in its own package (and you won't need it after all)